### PR TITLE
Add support for RISC-V 64-bit architecture

### PR DESCRIPTION
**Pull request by mitya57**
This is based on Ubuntu's patch by William Grant, that was added in <https://launchpad.net/ubuntu/+source/qtwebkit-opensource-src/5.212.0~alpha4-1ubuntu1>.

The most controversial change here is adding the `THREADS_PREFER_PTHREAD_FLAG`. But I can make it conditional or drop it completely if you want.

### CODE REVIEW

**Comment by mitya57** [comment]
```
JavaScriptCore.lib(JITOperations.cpp.obj) : error LNK2019: unresolved external symbol getHostCallReturnValue referenced in function handleHostCall

JavaScriptCore.lib(LLIntSlowPaths.cpp.obj) : error LNK2001: unresolved external symbol getHostCallReturnValue

bin\jsc.exe : fatal error LNK1120: 1 unresolved externals
```

This is unrelated to my changes, right?

File: Source/JavaScriptCore/CMakeLists.txt [hunk 1]
```diff
@@ -1289,6 +1289,7 @@ elseif (WTF_CPU_MIPS)
 elseif (WTF_CPU_SH4)
 elseif (WTF_CPU_X86)
 elseif (WTF_CPU_X86_64)
+elseif (WTF_CPU_RISCV64)
```
Comment (annulen) [inline Source/JavaScriptCore/CMakeLists.txt:1292]: This change breaks build with MSVC

**Comment by annulen** [comment]
>The most controversial change here is adding the THREADS_PREFER_PTHREAD_FLAG
It was considered by upstream (see https://bugs.webkit.org/show_bug.cgi?id=182622) but reverted because it caused issues in other cases

>This is unrelated to my changes, right?
It is directly caused by them, though easy to fix

Anyway I would prefer if such patch was merged to upstream first.

**Comment by annulen** [comment]
Note that you can pass THREADS_PREFER_PTHREAD_FLAG via cmake arguments (like -DTHREADS_PREFER_PTHREAD_FLAG=ON)

**Comment by mitya57** [comment]
Hi @annulen!

With upstream master, this patch is not needed, presumably because of 861472b7b939871bd622e85b26c4f3d7e5c60a7e. The only needed change is `THREADS_PREFER_PTHREAD_FLAG`, but as you say, one may set it on the command line.

I will wait for the CI to pass and then retarget to webkit-stable.

**Comment by annulen** [comment]
OK. Since this patch is not needed in `qtwebkit-dev` (which has aforementioned patches), it should go to `qtwebkit-5.212`, not `qtwebkit-stable`.

**Comment by mitya57** [comment]
That's where it is directed now, I just got confused by the message from Travis.

**Pull request merged**